src/seeds.quests.cpp

```diff
@@ -0,0 +1,1554 @@
+  m_documentGraph.eraseDocument(milestone_hash, true);
+
+}
+
+ACTION quests::activate (checksum256 quest_hash) {
+
+  hypha::Document quest_doc(get_self(), quest_hash);
+  hypha::Document quest_v_doc = get_variable_node_or_fail(quest_doc);
+
+  name creator = quest_doc.getCreator();
+
+  require_auth(creator);
+  check_type(quest_doc, graph::QUEST);
+
+  hypha::ContentWrapper quest_cw = quest_doc.getContentWrapper();
+  hypha::ContentWrapper quest_v_cw = quest_v_doc.getContentWrapper();
+
+  asset amount = quest_cw.getOrFail(FIXED_DETAILS, AMOUNT) -> getAs<asset>();
+  name fund = quest_cw.getOrFail(FIXED_DETAILS, FUND) -> getAs<name>();
+
+  check_quest_status_stage(quest_v_cw, ""_n, quest_stage_staged, "quests: can not activate the quest");
+  validate_milestones(quest_hash);
+
+  if (fund != creator) {
+    hypha::Document account_creator_doc = get_account_info(creator, true);
+    hypha::Edge::write(get_self(), get_self(), quest_hash, account_creator_doc.getHash(), graph::VALIDATE);
+
+    update_node(&quest_v_doc, VARIABLE_DETAILS, {
+      hypha::Content(STAGE, quest_stage_proposed)
+    });
+
+    propose_aux(quest_hash, creator, name("propactivate"), name("notactivate"));
+    return;
+  }
+
+  hypha::Document account_creator_doc = get_account_info(creator, false);
+  hypha::Document account_creator_v_doc = get_variable_node_or_fail(account_creator_doc);
+  hypha::ContentWrapper account_creator_v_cw = account_creator_v_doc.getContentWrapper();
+
+  asset staked_amount = account_creator_v_cw.getOrFail(VARIABLE_DETAILS, ACCOUNT_BALANCE) -> getAs<asset>();
+  asset locked_balance = account_creator_v_cw.getOrFail(VARIABLE_DETAILS, LOCKED_BALANCE) -> getAs<asset>();
+
+  check(staked_amount >= amount, "quests: not enough balance to activate the quest");
+
+  update_node(&account_creator_v_doc, VARIABLE_DETAILS, {
+    hypha::Content(ACCOUNT_BALANCE, staked_amount - amount),
+    hypha::Content(LOCKED_BALANCE, locked_balance + amount)
+  });
+
+  update_node(&quest_v_doc, VARIABLE_DETAILS, {
+    hypha::Content(STAGE, quest_stage_active)
+  });
+
+  print("ACTIVATE action (", creator, ") executed successfully\n");
+
+}
+
+ACTION quests::propactivate (checksum256 quest_hash) {
+
+  require_auth(get_self());
+
+  hypha::Document quest_doc(get_self(), quest_hash);
+  hypha::Document quest_v_doc = get_variable_node_or_fail(quest_doc);
+
+  update_node(&quest_v_doc, VARIABLE_DETAILS, {
+    hypha::Content(STAGE, quest_stage_active)
+  });
+
+  print("PROP ACTIVATE action executed successfully\n");
+
+}
+
+ACTION quests::notactivate (checksum256 quest_hash) {
+
+  hypha::Document quest_doc(get_self(), quest_hash);
+  hypha::Document quest_v_doc = get_variable_node_or_fail(quest_doc);
+
+  require_auth(get_self());
+
+  hypha::ContentWrapper quest_v_cw = quest_v_doc.getContentWrapper();
+  name stage = quest_v_cw.getOrFail(VARIABLE_DETAILS, STAGE) -> getAs<name>();
+  if (stage != quest_stage_proposed) { return; }
```

Comment:
Amm I don't think so, because this action is supposed to be used by the contract inside a proposal, so that should never happen, but I felt better adding it